actors/evm/src/interpreter/instructions/lifecycle.rs

```diff
@@ -1,17 +1,156 @@
-    _platform: &'r System<'r, BS, RT>,
-    _create2: bool,
+    state: &mut ExecutionState,
+    platform: &'r mut System<'r, BS, RT>,
+) -> Result<(), StatusCode> {
+    const CREATE_METHOD_NUM: u64 = 2;
+    let ExecutionState { stack, memory, .. } = state;
+
+    // Overall future work / TODOs:
+    //  readonly state things | ISSUE NEEDED
+    //  preload state items (~eq to EVM access list) | ISSUE NEEDED
+
+    #[derive(Serialize_tuple, Deserialize_tuple)]
+    struct CreateParams {
+        #[serde(with = "strict_bytes")]
+        code: Vec<u8>,
+        nonce: u64,
+    }
+
+    let value = stack.pop();
+    let (offset, size) = (stack.pop(), stack.pop());
+
+    let value = TokenAmount::from(&value);
+    let input_region =
+        get_memory_region(memory, offset, size).map_err(|_| StatusCode::InvalidMemoryAccess)?;
+
+    let input_data = if let Some(MemoryRegion { offset, size }) = input_region {
+        &memory[offset..][..size.get()]
+    } else {
+        return Err(StatusCode::ActorError(ActorError::assertion_failed(
+            "inicode not in memory range".to_string(),
+        )));
+    };
+
+    // bump nonce and flush state TODO before send
+    let mut nonce = 0;
+    platform.rt.transaction(|state: &mut State, _rt| {
+        // this may be redundant if we are compiling with checked integer math
+        state.nonce = state.nonce.checked_add(1).unwrap();
+        nonce = state.nonce;
+        Ok(())
+    })?;
+
+    let params = CreateParams { code: input_data.to_vec(), nonce };
+
+    create_init(stack, platform, RawBytes::serialize(&params)?, CREATE_METHOD_NUM, value)
+}
+
+pub fn create2<'r, BS: Blockstore, RT: Runtime<BS>>(
+    state: &mut ExecutionState,
+    platform: &'r mut System<'r, BS, RT>,
 ) -> Result<(), StatusCode> {
-    todo!()
+    const CREATE2_METHOD_NUM: u64 = 3;
+    let ExecutionState { stack, memory, .. } = state;
+
+    // see `create()` overall TODOs
+
+    #[derive(Serialize_tuple, Deserialize_tuple)]
+    struct Create2Params {
+        #[serde(with = "strict_bytes")]
+        code: Vec<u8>,
+        salt: [u8; 32],
+    }
+
+    let endowment = stack.pop();
+    let (offset, size) = (stack.pop(), stack.pop());
+    let salt = stack.pop();
+
+    let endowment = TokenAmount::from(&endowment);
+    let input_region =
+        get_memory_region(memory, offset, size).map_err(|_| StatusCode::InvalidMemoryAccess)?;
+
+    // BE encoded array
+    let salt: [u8; 32] = salt.into();
+
+    let input_data = if let Some(MemoryRegion { offset, size }) = input_region {
+        &memory[offset..][..size.get()]
+    } else {
+        return Err(StatusCode::ActorError(ActorError::illegal_argument(
+            "initcode not in memory range".to_string(),
+        )));
+    };
+    let params = Create2Params { code: input_data.to_vec(), salt };
+
+    platform.rt.transaction(|state: &mut State, _rt| {
+        // this may be redundant if we are compiling with checked integer math
```

Comment:
Better to double check. Rust should optimize that away.

(I mean, we should never hit this in practice anyways)

Reply:
Like, literally impossible?